authentik/events/signals.py

```diff
@@ -64,6 +68,26 @@ def on_user_write(sender, request: HttpRequest, user: User, data: dict[str, Any]
     data["created"] = kwargs.get("created", False)
     Event.new(EventAction.USER_WRITE, **data).from_http(request, user=user)
 
+    """Preserve the user's old password if UniquePasswordPolicy is enabled anywhere"""
+    user_changed_own_password = (
+        any("password" in x for x in data.keys())
+        and request.user.pk == user.pk
+        and SESSION_KEY_IMPERSONATE_USER not in request.session
+    )
+    if user_changed_own_password:
+        unique_password_policies = UniquePasswordPolicy.objects.all()
+
+        unique_pwd_policy_binding = PolicyBinding.objects.filter(
+            policy__in=unique_password_policies
+        ).filter(enabled=True)
+
```

Comment:
I reuse this query pattern a few times in different contexts. Would it be better to move this into a Manager method on `PolicyBinding`? Something like:

```python
PolicyBinding.objects.all_in_use(type=POLICY_MODEL_HERE)
```

Reply:
I think it does make sense to do this indeed.